### PR TITLE
Add ArcLayer example to the gallery

### PR DESCRIPTION
**Pull request by kaushikb11**
For #2968 

- Added ArcLayer demo example to the gallery.

### CODE REVIEW

**Comment by kaushikb11** [comment]
Screenshot from 2019-04-13 15-10-46

**Comment by xintongxia** [comment]
This is a great start. A few suggestions,

1. format the code as other examples. I am not sure it is because of github or not, but the continuation indent should be 2 spaces, also there should be an extra line after the last line of the file. I have an issue to track applying lint rule to our gallery examples.

2. could divide the `fetch` callback into small functions to make the code more readable.

```

const deckgl = new DeckGL({
  ...
}); // intialize deckgl

function renderLayers(data, selectedFeature) {
  selectedFeature = selectedFeature || data.features.find(f => f.properties.name === 'Los Angeles, CA') // default to feature with name 'Los Angeles, CA';


   // construct the layers
   // then in your GeoJsonLayer, 'onClick' callback could trigger renderLayers
   ...

   deckgl.setProps({layers: ...})

}

fetch(data-url)
  .then(...)
  .then(data => {renderLayers(data)});

```

**Comment by kaushikb11** [comment]
@xintongxia Did the required changes. Refactored the functions and fixed the lint errors.

File: examples/gallery/src/arc-layer.html [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    function renderLayers(data, selectedFeature) {
+
+      selectedFeature = selectedFeature || data.features.find(f => f.properties.name === 'Los Angeles, CA');
+      const arcLayer = getArcLayer(data, selectedFeature);
+      const countyLayer = new GeoJsonLayer({
+        id: 'geojson',
+        data,
+        stroked: false,
+        filled: true,
+        autoHighlight: true,
```
Comment (xintongxia) [inline examples/gallery/src/arc-layer.html]: disable this `autoHighlight`, not necessary.

Comment (kaushikb11) [reply examples/gallery/src/arc-layer.html]: But it helps the user in a way that when the user hovers over the map by a mouse pointer, he or she would know by the `highlighted` color that the `Arcs` could be generated by that part when clicked on it.

**Pull request merged**